geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/CacheClientNotifier.java

```diff
@@ -514,21 +514,16 @@ private CacheClientProxy registerClient(Socket socket, ClientProxyMembershipID p
       CacheClientProxy staleClientProxy = this.getClientProxy(proxyId);
       boolean toCreateNewProxy = true;
       if (staleClientProxy != null) {
-        if (staleClientProxy.isConnected() && staleClientProxy.getSocket().isConnected()) {
+        // Ping the proxy to double check if it's still connected
+        // If the proxy is still alive, that means there're 2 cache servers, should
+        // reject the incoming request to create a new proxy from the same client.
+        // If the client was reconnecting, the ping should fail because socket
+        // was closed, then it should create a new proxy instead.
+        ClientMessage message = new ClientPingMessageImpl();
+        staleClientProxy.sendMessageDirectly(message);
+        if (staleClientProxy.isAlive()) {
           successful = false;
           toCreateNewProxy = false;
-        } else {
```

Comment:
Hi Naba, would you please explain why this code is being removed?

Reply:
Because when ping failed, it will automatically do the remove.